src/v/cluster_link/service.cc

```diff
@@ -187,6 +188,28 @@ class kafka_consumer_groups_router : public consumer_groups_router {
     ss::sharded<kafka::group_router>* _router;
 };
 
+class health_monitor_based_partition_metadata_provider
+  : public partition_metadata_provider {
+public:
+    explicit health_monitor_based_partition_metadata_provider(
+      ss::sharded<cluster::health_monitor_frontend>* hm_frontend)
```

Comment:
nitpick: pass by reference?

Reply:
i see the convention in the cluster link directory is that we pass by pointer, happy to change that if you think i should